android/java/org/chromium/chrome/browser/app/domain/PendingItemsModel.java

```diff
@@ -0,0 +1,102 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.chromium.chrome.browser.app.domain;
+
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.MutableLiveData;
+import androidx.lifecycle.Transformations;
+
+import org.chromium.brave_wallet.mojom.AccountInfo;
+import org.chromium.brave_wallet.mojom.BraveWalletConstants;
+import org.chromium.brave_wallet.mojom.BraveWalletService;
+import org.chromium.brave_wallet.mojom.CoinType;
+import org.chromium.brave_wallet.mojom.JsonRpcService;
+import org.chromium.brave_wallet.mojom.KeyringService;
+import org.chromium.brave_wallet.mojom.TransactionInfo;
+import org.chromium.brave_wallet.mojom.TransactionStatus;
+
+import org.chromium.chrome.browser.crypto_wallet.util.PendingTxHelper;
+import org.chromium.chrome.browser.crypto_wallet.util.Utils;
+
+
+public class PendingItemsModel {
+    private KeyringService mKeyringService;
+    private JsonRpcService mJsonRpcService;
+    private BraveWalletService mBraveWalletService;
+    private CryptoModel mCryptoModel;
+    private PendingTxHelper mPendingTxHelper;
+    private final MutableLiveData<Boolean> _mWalletIconNotificationVisible =
+            new MutableLiveData<>(false);
+    public final LiveData<Boolean> mWalletIconNotificationVisible = _mWalletIconNotificationVisible;
+
+    public PendingItemsModel(KeyringService mKeyringService, JsonRpcService mJsonRpcService,
+            BraveWalletService mBraveWalletService, PendingTxHelper mPendingTxHelper) {
+        this.mKeyringService = mKeyringService;
+        this.mBraveWalletService = mBraveWalletService;
+        this.mJsonRpcService = mJsonRpcService;
+        this.mPendingTxHelper = mPendingTxHelper;
+    }
```

Comment:
There should be a `resetServices` method to handle https://github.com/brave/brave-core/blob/495775f32cb752a96e189d8b9b9d5c6208ec06b7/android/java/org/chromium/chrome/browser/app/BraveActivity.java#L1315
as the services can be reinitialised in case of an error then the services in this class won't be updated

Reply:
yeah, that's a must do part, otherwise we will have crashes on accessing them